src/main/java/ro/developmentfactory/thedrones/controller/DroneControler.java

```diff
@@ -0,0 +1,40 @@
+package ro.developmentfactory.thedrones.controller;
+
+import jakarta.validation.Valid;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+import ro.developmentfactory.thedrones.entity.Drone;
+import ro.developmentfactory.thedrones.service.DroneServcie;
+
+import java.util.List;
+
+@RestController
+public class DroneControler {
```

Comment:
I see that you are using `Drone` object which is defined as your database entity as an input and output for the controller. this is not a good choice as:
* we don't want our database structure to be dependent on the input / output
* we don't want input / output to depend on the database structure
* we donn't want to allow API users to set the values (ids, created_at, updated_at) that in fact will not be saved to the database - this is very confusing
* we don't want to tell the rest of the world (because we treat API as a public specification) know the details of our database design; this is even a security concern.

for this, I would recommend creating `DroneRequest` and `DroneResponse` classes and using those.

Reply:
done